### PR TITLE
122 frontend spotify component

### PR DESCRIPTION
**Pull request by BitsyBirb**
# What type of PR is this? (Check all that apply)

- [x] ✨**feature**: Introduces completely new code or new features.
- [ ] 🐛**fix**: Implements changes that fix a bug. Ideally, reference an issue if present.
- [ ] ♻️**refactor**: Includes any code-related change that is neither a fix nor a feature.
- [ ] ✅**build**: Encompasses all changes related to the build of the software, including changes to dependencies or the addition of new ones.
- [ ] ⚡️**test**: Pertains to all changes regarding tests, whether adding new tests or modifying existing ones.
- [ ] 🚰**ci**: Involves all changes related to the configuration of continuous integration, such as GitHub Actions or other CI systems.
- [ ] 📚**docs**: Includes all changes to documentation, such as README files, or any other documentation present in the repository.
- [ ] 🗑️**chore**: Captures all changes to the repository that do not fit into the above categories.

# Description

4 Files Changed:
   `apps/expo/src/app/nav/index.tsx` and `apps/expo/src/app/spotify/index.tsx`: Created a page to see the new widget.
   `apps/expo/src/components/spotify/Spotify.tsx`: Makes up the bulk of the widget. Image + song/album info.
   `apps/expo/src/components/spotify/song_progress.tsx`: Helper component to display times as Min:Sec and progress bar.

# Tests

How was this tested?

- [ ] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [x] Manual tests
- [ ] Tests were NOT needed
- [ ] Other (explain below)

Explanation:
Just grabbed a row of data from the database and made sure values/output looked fine. Refer to attached screenshots.

# Documentation

- [ ] Added to README.me
- [ ] Seperate document
- [x] NO documentation needed

Link to external documentation:

# [Optional] Screenshots
image

image

image

# [Optional] Are there any post-deployment tasks we need to perform?
No

# Additional notes
We will very likely need to update the styling and appearance to better fit our app. The current widget is rather primitive and not fleshed-out.

### CODE REVIEW

**Comment by BitsyBirb** [comment]
Note on testing:

In order to test the component, there are two main cases based on retrieved data using userID in `apps/expo/src/app/spotify/index.tsx`.

General testing will go something along these lines:

1) Make sure a SpotifyData row exists for the user you will be querying.
2) Use that user's ID to query the SpotifyData using the API (just update the ID field in `apps/expo/src/app/spotify/index.tsx` from above.
3) Depending on if the row itself had isPlaying as false or true, one of two things will happen:
3a) If it's false, then the third screenshot should show.
3b) Otherwise something similar to the first two screenshots will show.

To check what shows up, navigate to "Screens" and click on "Spotify" within the mobile app itself.
Results should be rendered there.

**Comment by thienguen** [comment]
Can't test the api
@BitsyBirb 

look here

https://github.com/UNLV-CS472-672/2024-S-GROUP3-Barbell/tree/122-frontend-spotify-component/packages/api/src

image


1. you are behind main 15 commits (sync it)
2. this branch specifically doesn't have your spotify.ts ? API 

image

1. how did you have this? me confused af
2. and it does error out here.

**Comment by BitsyBirb** [comment]
So on my local branch I just copied the code for the API (which probably wasn't ideal) as it hadn't been approved yet. It's using the code from that API but I can merge and update the PR if you'd like.

**Comment by thienguen** [comment]
> @thienguen So on my local branch I just copied the code for the API (which probably wasn't ideal) as it hadn't been approved yet. It's using the code from that API but I can merge and update the PR if you'd like.

yes please do so, 

```
git commit -m 'minor changes'  <-- fresh your current branch
git checkout main
git pull <-- get latest main
git checkout "your branch"
git merge --no-ff main <-- sync with main
```

if there is any conflict, resolve them, doing so you won't have to copy the API over

image

File: apps/expo/src/components/spotify/Spotify.tsx
Comment (thienguen) [inline apps/expo/src/components/spotify/Spotify.tsx:6]: That's good, install Prettier for vscode and format this file.

Now you have 2 options before i hit approve and merge in this pr

1. Have unit test for the spotfify component you wrote, its could be a bit tedious but I'll be on call with you

2. leave the unit tests on another beautiful day, just format it and push a new commits with the changes, I'll hit approve and merge for you from there 

lgtm

**Review by thienguen — CHANGES_REQUESTED** [review]

File: apps/expo/src/app/spotify/index.tsx
Comment (elliotwesoff) [inline apps/expo/src/app/spotify/index.tsx:6]: 😂😂😂 see: "how to center a div"

File: apps/expo/src/components/spotify/Spotify.tsx
Comment (elliotwesoff) [inline apps/expo/src/components/spotify/Spotify.tsx:32]: Good markdown formatting here. brownie points for you

File: apps/expo/src/components/spotify/song_progress.tsx
Comment (elliotwesoff) [inline apps/expo/src/components/spotify/song_progress.tsx:47]: If @thienguen merged his color.ts changes to main, could you please sync with main and use his values? i.e. we shouldn't hard code any color hex codes. Just do something like `import { colors } from '~/colors'; ... { backgroundColor: colors.lightGrey }`

**Review by elliotwesoff — CHANGES_REQUESTED** [review]
Just fix the hard coded color string in `song_progress.tsx` and you've got my approval.

**Comment by thienguen** [comment]
I'll make a commit, here to accommodate the changes. @elliotwesoff 

Also, we do have color.ts, but ultimately we will try to use tailwind? so the color. ts will be delete soon (you can see marcos hate on styleSheet LMAFO) 🤣

But soon, @BitsyBirb starting after this pr, you will need to write tailwind.

can you give this approval Elliot?

**Review by elliotwesoff — APPROVED** [review]

File: apps/expo/src/components/spotify/Spotify.tsx
Comment (thienguen) [reply apps/expo/src/components/spotify/Spotify.tsx:32]: my commit will accomodates the format🤣

File: apps/expo/src/components/spotify/song_progress.tsx
Comment (thienguen) [reply apps/expo/src/components/spotify/song_progress.tsx:47]: switching to native wind, erm...i will present on tuesday, this is worth brough up

**Review by thienguen — APPROVED** [review]

**Comment by thienguen** [comment]
approved

**Pull request merged**
